test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -111,6 +111,31 @@ session_simulator::rollback_transaction(const std::string &config)
     return (0);
 }
 
+int
+session_simulator::prepare_transaction(const std::string &config)
+{
+    /* Make sure that the transaction from this session is running. */
+    if (!_txn_running)
+        WT_SIM_RET_MSG(EINVAL, "'prepare_transaction' only permitted in a running transaction");
+
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    std::map<std::string, std::string> config_map;
+
+    const std::vector<std::string> supported_ops = {"prepare_timestamp"};
+    const std::vector<std::string> unsupported_ops;
+
+    WT_SIM_RET_MSG(ts_manager->parse_config(config, supported_ops, unsupported_ops, config_map),
+      "Incorrect config (" + config + ") passed in prepare_transaction");
+
+    auto pos = config_map.find("prepare_timestamp");
```

Comment:
Are there no rules for the prepare timestamp?

Reply:
This will be done in WT-9232, I will add a fix me comment for now.

Reply:
Rules for the prepared transaction is added to this PR, however, as mentioned earlier that the rules for prepared timestamp will be added in WT-9232